src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4517,15 +4519,91 @@ public void testCreateServiceWithDefaults() throws Exception {
+            .name("asecret")
+            .data(secretData)
+            .labels(labels)
+            .build();
+
+    final SecretCreateResponse secretResponse = sut.createSecret(secretSpec);
+    final String secretId = secretResponse.id();
+    assertThat(secretId, is(notNullValue()));
+
+    final SecretFile secretFile = SecretFile.builder()
+            .name("bsecret")
+            .build();
+    final SecretBind secretBind = SecretBind.builder()
+            .file(secretFile)
+            .secretId(secretId)
+            .secretName("asecret")
+            .build();
+
+    final String[] commandLine = {"ping", "-c4", "localhost"};
+    final TaskSpec taskSpec = TaskSpec
+            .builder()
+            .containerSpec(ContainerSpec.builder().image("alpine")
+                    .secrets(Arrays.asList(secretBind))
+                    .hostname(hostname)
+                    .hosts(Arrays.asList(hosts))
+                    .healthcheck(Healthcheck.create(Arrays.asList(healthcheckCmd), 30L, 3L, 3))
+                    .command(commandLine).build())
+            .build();
+    final String serviceName = randomName();
+    final ServiceSpec spec = ServiceSpec.builder()
+            .name(serviceName)
+            .taskTemplate(taskSpec)
+            .build();
+
+    final ServiceCreateResponse response = sut.createService(spec);
+
+    final Service service = sut.inspectService(response.id());
+
+    assertThat(service.spec().name(), is(serviceName));
+    final Matcher<String> imageMatcher = dockerApiVersionLessThan("1.25")
+            ? is("alpine") : startsWith("alpine:latest@sha256:");
+    assertThat(service.spec().taskTemplate().containerSpec().image(), imageMatcher);
+    assertThat(service.spec().taskTemplate().containerSpec().hostname(), is(hostname));
+    assertThat(service.spec().taskTemplate().containerSpec().hosts(), containsInAnyOrder(hosts));
+    assertThat(service.spec().taskTemplate().containerSpec().secrets().size(),
+            equalTo(1));
+    assertThat(service.spec().taskTemplate().containerSpec().secrets().get(0).secretId(),
+            equalTo(secretId));
+    assertThat(service.spec().taskTemplate().containerSpec().secrets().get(0).file().name(),
+            equalTo("bsecret"));
+    assertThat(service.spec().taskTemplate().containerSpec().healthcheck().test(),
+            equalTo(Arrays.asList(healthcheckCmd)));
+    assertThat(service.spec().taskTemplate().containerSpec().healthcheck().interval(),
+            equalTo(30L));
+    assertThat(service.spec().taskTemplate().containerSpec().healthcheck().timeout(),
+            equalTo(3L));
+    assertThat(service.spec().taskTemplate().containerSpec().healthcheck().retries(),
+            equalTo(3));
+  }
+
+  @Test
   public void testInspectService() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "swarm support");
 
     final String[] commandLine = {"ping", "-c4", "localhost"};
-    final String hostname = "tshost-{{.Task.Slot}}";
```

Comment:
"hostname" is specific to api versio 1.26, so I moved this test to testCreateServiceWithSecret. since testCreateServiceWithSecret is testing features specifc to 1.2.6